scripts/pi-hole/js/settings.js

```diff
@@ -140,6 +140,35 @@ $("#DHCPchk").click(function() {
 	$("#dhcpnotice").prop("hidden", !this.checked).addClass("lookatme");
 });
 
+function loadCacheInfo()
+{
+    $.getJSON("api.php?getCacheInfo", function(data) {
+        if("FTLnotrunning" in data)
+        {
+            return;
+        }
+
+        // Fill table with obtained values
+        $("#cache-size").text(parseInt(data["cacheinfo"]["cache-size"]));
+        $("#cache-inserted").text(parseInt(data["cacheinfo"]["cache-inserted"]));
+
+        // Highlight early cache removals when present
+        var cachelivefreed = parseInt(data["cacheinfo"]["cache-live-freed"]);
+        $("#cache-live-freed").text(cachelivefreed);
+        if(cachelivefreed > 0)
+        {
+            $("#cache-live-freed").parent("tr").addClass("lookatme");
+        }
+        else
+        {
+            $("#cache-live-freed").parent("tr").removeClass("lookatme");
+        }
+
+        // Update cache info every 10 seconds
+        setTimeout(loadCacheInfo, 10000);
```

Comment:
Does the cache need to be refreshed so often? 10 minutes seems more reasonable.

Reply:
I wanted this information to update rather frequently such that new numbers are refreshed more often. For instance, the cache insertions count may change every second if new domains are queried. Also, this allows for a resized cache to be shown without refreshing the page.

As I don't think that users will stay on the Settings page for very long (and this is the only dynamically loaded content), I'd propose to either leave it like this or remove the refreshing after a certain timeout altogether (for slightly simpler code). I'm open for a discussion.

Reply:
How about increasing it to 30 seconds?